scripts/powershell/version-repo-with-tag.ps1

```diff
@@ -0,0 +1,43 @@
+<#
+.DESCRIPTION
+Tags latest or a specific commit in the specified Repo.
+Uses a combination of the Repo Name, Current Date and a Version number to create the tag.
+Tag serve as a means of versioning the entire repo.
+.PARAMETER RepoName
+The name of the repository. EG "azure-sdk-tools"
+.PARAMETER RepoOrg
+The owning organization of the repository. EG "Azure"
+.PARAMETER RepoType
+Should be either a github or git (Azure) repo
+.PARAMETER CommitSHA
+The commitSHA to tag. Defaults to master
+.PARAMETER RepoProjectID
+For git Repos supply the project ID. Default to the internal project ID set by the pipeline
+.PARAMETER PATForAPI
+Personal Access Token for authentication to the Repo
+#>
+param (
+    [Parameter(Mandatory = $true)]
+    [string] $RepoName,
+
+    [Parameter(Mandatory = $true)]
+    [string] $SourceDir,
+
+    [string] $CommitSHA="HEAD",
+
+    [Parameter(Mandatory = $true)]
+    [string] $BuildNumber
+)
+
+Write-Host $MyInvocation.Line
+
+$VersionTag = $RepoName + "_" + $BuildNumber
+
+pushd "$SourceDir/$RepoName"
+
+Write-Host "Get actual commit SHA"
+$CommitSHA = git rev-parse $CommitSHA
```

Comment:
If you are actually at the commit where you want to produce the tag you don't need to do this you can just call get tag and it will take the head commit.

Reply:
My thinking is that there might be c case where we want to tag an older commit. Say for example a servicing branch wants to use an older 'version'.

Reply:
Then we would queue the build from that sha/branch when we queue this build for tagging/releasing.